### PR TITLE
feat(patient): add appointment button

### LINKED ISSUE #1841: Add new appointment button in patient's appointments tab
## Scenario
**AS**
A user with the ability to create appointments

**I WANT**
A new appointment button in the patient's appointments tab so that 

**SO THAT**
I can quickly make a new appointment for that patient. 

> **GIVEN**
> A user with the ability to create an appointment and an existing patient
> 
> **WHEN**
> I click the new appointment button 
> 
> **THEN**
> I should be navigate to the new appointments screen with the patients box filled out with the name of the patient which i just came from

## Technical Notes
* Button should have add appointment icon

### PR DESCRIPTION
**Pull request by oizuldan**
Fixes #1841.

**Changes proposed in this pull request:**

- Add "Add Appointment" button on the patient page, appointments tab.

### CODE REVIEW

File: src/patients/appointments/AppointmentsList.tsx [hunk 1]
```diff
@@ -54,6 +54,20 @@ const AppointmentsList = (props: Props) => {
...
+        <div className="col-md-12 d-flex justify-content-end">
+          <Button
+            outlined
+            color="success"
+            icon="add"
+            iconLocation="left"
+            onClick={() => history.push('/appointments/new')}
+          >
+            {t('patient.appointments.new')}
+          </Button>
```
Comment (jackcmeyer) [inline src/patients/appointments/AppointmentsList.tsx:67]: need to add some tests for this: 

- [ ] button gets rendered w/ correct label
- [ ] when the button gets clicked, it navigates to the correct location

File: src/patients/appointments/AppointmentsList.tsx [hunk 2]
```diff
@@ -54,6 +54,20 @@ const AppointmentsList = (props: Props) => {
...
+      <div className="row">
+        <div className="col-md-12 d-flex justify-content-end">
+          <Button
+            outlined
+            color="success"
+            icon="add"
+            iconLocation="left"
+            onClick={() => history.push('/appointments/new')}
+          >
+            {t('patient.appointments.new')}
```
Comment (jackcmeyer) [inline src/patients/appointments/AppointmentsList.tsx:66]: Let's use the the existing `New Appointment` label that you can see on the appointments calendar.

let's also add the new appointment icon to the button

File: src/patients/appointments/AppointmentsList.tsx (hunk 1 above)
Comment (oizuldan) [reply src/patients/appointments/AppointmentsList.tsx]: ok

File: src/patients/appointments/AppointmentsList.tsx (hunk 2 above)
Comment (oizuldan) [reply src/patients/appointments/AppointmentsList.tsx]: ok

File: src/patients/appointments/AppointmentsList.tsx (hunk 1 above)
Comment (oizuldan) [reply src/patients/appointments/AppointmentsList.tsx]: Btw can you please tell how to run tests on my local machine?

Comment (jackcmeyer) [reply src/patients/appointments/AppointmentsList.tsx]: `yarn test`

**Comment by oizuldan** [comment]
@jackcmeyer I have added tests and everything seems to work fine

File: src/__tests__/patients/appointments/NewAppointment.test.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import '../../../__mocks__/matchMediaMock'
```
Comment (jackcmeyer) [inline src/__tests__/patients/appointments/NewAppointment.test.tsx:1]: I think this file name should be `src/__tests__/patients/appointments/AppointmentsList.test.tsx` to better reflect the component it is testing.

File: src/__tests__/patients/appointments/NewAppointment.test.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  )
+
+  return wrapper
+}
+
+describe('AppointmentsList', () => {
+  describe('add new appointment button', () => {
+    beforeEach(() => {
+      jest.resetAllMocks()
+      jest.spyOn(PatientRepository, 'saveOrUpdate')
```
Comment (jackcmeyer) [inline src/__tests__/patients/appointments/NewAppointment.test.tsx:52]: this mock is not necessary

**Comment by jackcmeyer** [comment]
Hey @oizuldan, let me know if you need any help getting this pull request finished up!

**Comment by oizuldan** [comment]
> Hey @oizuldan, let me know if you need any help getting this pull request finished up!

Okay!

**Comment by oizuldan** [comment]
> A stylistic change:
> 
> Can we make the New Appointment aligned with all of the other "Add" buttons in the diagnoses tab, allergies tab, etc.

Will do

**Comment by jackcmeyer** [comment]
@oizuldan code changes look good. just need to fix linter errors. let me know if you need any help!

**Comment by oizuldan** [comment]
> @oizuldan code changes look good. just need to fix linter errors. let me know if you need any help!

Do I need to fix these errors? (I am a little confused since you already approved my PR)

**Pull request merged**